### ISSUE TITLE
DOSXYZnrc source 20, number of control points

### ISSUE THREAD

**Issue by AndrewWAlexander** [nrc-cnrc/EGSnrc#230]
When there is only 1 control point for a source 20 run, the code runs, but returns garbage results. It would appear that source 20 cannot be used with only one control point. As such, it might be useful for the simulation to return an error when only 1 control point is defined. 

nset value must be > 1

**Comment by ftessier** [comment]
@AndrewWAlexander can you confirm if this issue is related to and resolve by pull request #223?

**Comment by AndrewWAlexander** [comment]
Sure I can do that

**Comment by marenaud** [comment]
I think the code with the "bug" is around [line 4146 of srcxyznrc.mortran](https://github.com/nrc-cnrc/EGSnrc/blob/develop/HEN_HOUSE/user_codes/dosxyznrc/srcxyznrc.mortran#L4145) for interpolating the source position parameters
```
   "calculate rotation matrix"                                                  
   " get theta, phi, phicol x|y|ziso"                                           
   theta(1)=thetatemp(K-1)+(thetatemp(K)-thetatemp(K-1))*                       
        (frMU_indx-muIndex(K-1))/(muIndex(K)-muIndex(K-1));                     
   phi(1)=phitemp(K-1)+(phitemp(K)-phitemp(K-1))*                               
        (frMU_indx-muIndex(K-1))/(muIndex(K)-muIndex(K-1));                     
   phicol=phicoltemp(K-1)+(phicoltemp(K)-phicoltemp(K-1))*                      
        (frMU_indx-muIndex(K-1))/(muIndex(K)-muIndex(K-1));                     
   xiso=xtemp(K-1)+(xtemp(K)-xtemp(K-1))*                                       
        (frMU_indx-muIndex(K-1))/(muIndex(K)-muIndex(K-1));                     
   yiso=ytemp(K-1)+(ytemp(K)-ytemp(K-1))*                                       
        (frMU_indx-muIndex(K-1))/(muIndex(K)-muIndex(K-1));                     
   ziso=ztemp(K-1)+(ztemp(K)-ztemp(K-1))*                                       
        (frMU_indx-muIndex(K-1))/(muIndex(K)-muIndex(K-1));     
```
which explicitly assumes that there is more than one control point defined instead of having a special case if ```nset = 1```, as the code:

```
   DO I=1, nset[
    "*********************************************************************"
    " get setting from randomly generated frMU_indx
    "*********************************************************************"
      IF(frMU_indx<muIndex(I))[
       K=I;
       EXIT;
      ]
   ]
```
will always set K=1 if nset == 1 (assuming muIndex of that one control point is 1.0), and the previous code is trying to address all those arrays with K-1 with undefined behaviour (I think?).

**Comment by ftessier** [comment]
@blakewalters can you look into this issue?

**Comment by blakewalters** [comment]
Sure.  In time for this release?

**Comment by ftessier** [comment]
No, for the develop branch. Although we can always merge in master if deemed necessary.

**Comment by AndrewWAlexander** [comment]
Thanks Marc for finding the source of the issue. I agree, the gui fixes in [#223 ](https://github.com/nrc-cnrc/EGSnrc/pull/223), is unrelated to this issue.
